### DISCUSSION TITLE
Laravel Config & Route Cache

### DISCUSSION THREAD

**Discussion by moisish** [serversideup/docker-php#187] [Q&A] [answered]
Previously I was using my own docker images and I was having different entrypoints based on what I needed to use the container for, queues, task, webserver

```
COPY infra/docker/entrypoints/webserver.sh /usr/bin/webserver
RUN chmod +x /usr/bin/webserver

COPY infra/docker/entrypoints/scheduler.sh /usr/bin/scheduler
RUN chmod +x /usr/bin/scheduler

COPY infra/docker/entrypoints/horizon.sh /usr/bin/horizon
RUN chmod +x /usr/bin/horizon
```

The horizon entrypoint was for example which was running the config:cache and route:cache first
These two commands cannot be run during the build of the image

```
php /var/www/html/artisan config:cache
php /var/www/html/artisan route:cache
/usr/bin/supervisord -c /etc/supervisor/supervisord.conf
```

I decided to switch to your images and use fpm-ngxin. How can I implement the equivalent ?

**Comment by jaydrogers** [comment, marked as answer]
Thanks for stopping by @moisish!

I have a Laravel automations script in that image (source located here: https://github.com/serversideup/docker-php/blob/main/src/fpm/etc/s6-overlay/scripts/laravel-automations)

I used to have `artisan config:cache` and `route:cache`, but it created strange behaviors with APIs in an older version of Laravel (pretty sure it was like 5.x).

I removed them because it was a core Laravel bug, but I am not sure if it has been resolved or not. I assume it has with how many enhancements have been made since then 😅

### To extend the image
I have some docs here on how to customize things: https://serversideup.net/open-source/docker-php/docs/guide/customizing-the-image

You'll likely want to add your own script that follows my Laravel automations script. [Be sure to NOT use supervisor](https://serversideup.net/open-source/docker-php/docs/getting-started/these-images-vs-others#s6-overlay-vs-supervisor) and follow the S6 Overlay methods to execute your script. See how I do this under the [fpm folder](https://github.com/serversideup/docker-php/tree/main/src/fpm) (which fpm-nginx builds off that base image).

Hope this helps!

**Reply by moisish** [reply]
Hello @jaydrogers 

First of all thanks for all your work on the images.

I have already my own image based on yours as I need to add more stuff. 
I am certainly not going to use supevisord anymore; i had my old entrypoint changed to 

```
php /var/www/html/artisan config:cache
php /var/www/html/artisan route:cache
php /var/www/html/artisan horizon
```

I am not familiar with s6 so I will need some more help with that. If I add my own `/etc/s6-overlay/scripts` on the docker file will that be picked up ?

**Reply by moisish** [reply]
Also, how can I customized which entrypoint I need to run?
I used to have the same container to run queues, scheduler and the app. 

My containers are running on AWS ECS and I was overriding the entrypoint as 
```
          Command:
            - "/bin/sh"
            - "-c"
            - "/usr/bin/webserver"
       ```

```
          Command:
            - "/bin/sh"
            - "-c"
            - "/usr/bin/scheduler"
       ```

**Reply by jaydrogers** [reply]
The S6 docs are an excellent place to get started: https://github.com/just-containers/s6-overlay#writing-a-service-script

Between reading the S6 docs, and seeing how I organized my `laravel-automations` script, you should be able to add an additional script that will execute after mine 👍

**Reply by moisish** [reply]
Thanks @jaydrogers 

I have added my own script for laravel caching and I have submited a PR with enhancements on your laravel-automations.

Regards my current entrypoints to run different laravel services, what is the best way to implement that now?

Can an s6 script be called at runtime? Should I call my bash script as before?
